src/executioners/OptimizeSolve.C

```diff
@@ -263,16 +261,47 @@ OptimizeSolve::objectiveAndGradientFunctionWrapper(
 }
 
 PetscErrorCode
-OptimizeSolve::hessianFunctionWrapper(Tao /*tao*/, Vec x, Mat hessian, Mat /*pc*/, void * ctx)
+OptimizeSolve::hessianFunctionWrapper(Tao /*tao*/, Vec x, Mat /*hessian*/, Mat /*pc*/, void * ctx)
 {
   auto * solver = static_cast<OptimizeSolve *>(ctx);
```

Comment:
I think this entire function can be empty.  I don't think that it is doing anything that hasn't already been done by the first MatShellSetOperation on ln 130-134.  This wrapper is setting x into the solvers _parameters but those _parameters are never used in the hessian wrappers or functions.  Should these _parameters be used in applyHessianWrapper?

Reply:
Yes, let's empty out this function. Eventually we can assemble an actual Hessian as a preconditioner here.

Reply:
Kept the function and moved MatShellSetOperation into this routine.